skypeweb/skypeweb_messages.c

```diff
@@ -1319,12 +1322,21 @@ skypeweb_set_status(PurpleAccount *account, PurpleStatus *status)
 void
 skypeweb_set_idle(PurpleConnection *pc, int time)
 {
+	const gchar *status_id;
+	PurpleStatus *status;
+
 	SkypeWebAccount *sa = purple_connection_get_protocol_data(pc);
 	
-	if (time < 30) {
-		skypeweb_set_statusid(sa, "Online");
-	} else {
-		skypeweb_set_statusid(sa, "Idle");
+	status = purple_account_get_active_status(purple_connection_get_account(pc));
+	status_id = purple_status_get_id(status);
+
+	/* Only go idle if active status is online  */
+	if (!strcmp(status_id, SKYPEWEB_STATUS_ONLINE)) {

```

Comment:
Preferred to use `purple_str_equal()` instead of `strcmp()` (for readability and NULL checks)